### PR TITLE
Detect component from general plan during import

### LINKED ISSUE #607: tmt test import --nitrate should detect general plans and fill component attribute accordingly
Component attribute [specification](https://tmt.readthedocs.io/en/stable/spec/tests.html#component) says that

> Relevant fedora/rhel source package names
> It’s useful to be able to easily select all tests relevant for given component or package. As they do not always have to be stored in the same repository and because many tests cover multiple components a dedicated field is needed.

In Nitrate we use presence General plan for this.

If import reads Nitrate it should process this information as well.

### PR DESCRIPTION
**Pull request by lukaszachy**
Disabled by default, use --general to enable.
Complementary to export --general which keeps only Genereral plans for
specified components

Fix #607

### CODE REVIEW

**Comment by lukaszachy** [comment]
I've realized that components from general plan should override components from test case (which was set by tcms-submit according to the makefile). 
UC1 - Intergration test was linked to another component's General plan without updating makefile/case components
UC2 - Test was cloned for another component and old components were intact (they are not actively used by current tooling)

For both cases possible `tmt test export --nirate --general` will remove more general plans than it should

**Review by hegerj — CHANGES_REQUESTED** [review]

File: tmt/convert.py [hunk 1]
```diff
@@ -687,8 +696,21 @@ def read_nitrate_case(testcase, makefile_data=None):
         echo(style('tier: ', fg='green') + data['tier'])
     except KeyError:
         pass
-    # Component
-    data['component'] = [comp.name for comp in testcase.components]
+    # Detect components either from general plans
+    if general:
+        data['component'] = []
+        for nitrate_plan in [plan for plan in testcase.testplans]:
```
[the remark this answers is not in the document]
Comment (lukaszachy) [reply tmt/convert.py:702]: good catch

**Comment by lukaszachy** [comment]
List comprehension removed and rebased to the current main branch.

**Review by hegerj — APPROVED** [review]
LGTM

**Review by psss — APPROVED** [review]
Looks fine, thanks for the improvement. What about using a non-white-space match `r"(\S+?)\s*/\s*General"` for the component name instead of anything to ignore possible spaces at the beginning?

**Comment by lukaszachy** [comment]
non-white-space matching looks better.

**Pull request merged**